qubesmanager/qube_manager.py

```diff
@@ -1042,43 +1042,50 @@ def action_killvm_triggered(self):
                         "ERROR: {1}").format(vm.name, ex))
                 return
 
+    def open_settings(self, vm, tab='basic'):
+        with common_threads.busy_cursor():
+            settings_window = settings.VMSettingsWindow(
+                vm, self.qt_app, tab)
+        settings_window.exec_()
+
+        vm_deleted = False
+
+        try:
+            # the VM might not exist after running Settings - it might
+            # have been cloned or removed
+            self.vms_in_table[vm.qid].update()
+        except exc.QubesException as ex:
+            QtWidgets.QMessageBox.warning(
+                self,
+                self.tr("Qube settings unavailable"),
+                self.tr(
+                    "Qube settings cannot be opened. The qube might have "
+                    "been removed or unavailable due to policy settings."
+                    "\nError: {}".format(str(ex))))
```

Comment:
Does this cover the case of removing/renaming VM (from within the settings window)? If so, it isn't an error. I think you need to catch an error and show this message if earlier part of this function fails (either constructing the object, or exec_() call - not sure.